opengrok-indexer/src/main/java/org/opengrok/indexer/configuration/RuntimeEnvironment.java

```diff
@@ -1996,14 +1843,45 @@ public File getDtagsEftar() {
     }
 
     public SuggesterConfig getSuggesterConfig() {
-        return (SuggesterConfig) getConfigurationValue("suggesterConfig");
+        return syncReadConfiguration(Configuration::getSuggesterConfig);
     }
 
     public void setSuggesterConfig(SuggesterConfig config) {
-        setConfigurationValue("suggesterConfig", config);
+        syncWriteConfiguration(Configuration::setSuggesterConfig, config);
+    }
+
+    /**
+     * Applies the specified function to the runtime configuration, after having
+     * obtained the configuration read-lock (and releasing afterward).
+     * @param function a defined function
+     * @param <R> the type of the result of the function
+     * @return the function result
+     */
+    public <R> R syncReadConfiguration(Function<Configuration, R> function) {
+        try (ResourceLock resourceLock = configLock.readLockAsResource()) {
+            //noinspection ConstantConditions to silence unreference auto-closeable
+            assert resourceLock != null;
+            return function.apply(configuration);
+        }
+    }
+
+    /**
+     * Performs the specified operation which is provided the runtime
+     * configuration and the specified argument, after first having obtained the
+     * configuration write-lock (and releasing afterward).
+     * @param consumer a defined consumer
+     * @param v the input argument
+     * @param <V> the type of the input to the operation
+     */
+    public <V> void syncWriteConfiguration(ConfigurationValueConsumer<V> consumer, V v) {
+        try (ResourceLock resourceLock = configLock.writeLockAsResource()) {
+            //noinspection ConstantConditions to silence unreference auto-closeable
+            assert resourceLock != null;
```

Comment:
why do you assert this when you're explicitely checking this in write|readLockAsResource?

Reply:
That's to silent an annoying Java warning of the auto-closeable not being referenced in the block which was a blocker in IDEA when it runs tests with warnings-as-errors

Reply:
what about `@SuppressWarnings("unused")` ?

Reply:
No it’s a very focused warning for auto-closeable